src/request_handler.cc

```diff
@@ -48,22 +48,24 @@ RequestHandler::RequestHandler(std::shared_ptr<ContentManager> content)
 {
 }
 
-void RequestHandler::splitUrl(const std::string& url, char separator, std::string& path, std::string& parameters)
+std::tuple<std::string_view, std::string_view> RequestHandler::splitUrl(std::string_view url, char separator)
 {
-    const auto i1 = std::size_t { [=]() {
-        if (separator == '/')
-            return url.rfind(separator);
-        if (separator == '?')
-            return url.find(separator);
+    std::size_t i1 = 0;
+    switch (separator) {
+    case '/':
+        i1 = url.rfind('/');
+        break;
+    case '?':
+        i1 = url.find('?');
+        break;
+    default:
         throw_std_runtime_error("Forbidden separator: {}", separator);
-    }() };
+    }
 
     if (i1 == std::string::npos) {
-        path = url;
-        parameters.clear();
+        return { url, std::string_view() };
     } else {
-        parameters = url.substr(i1 + 1);
-        path = url.substr(0, i1);
+        return { url.substr(0, i1), url.substr(i1 + 1) };
```

Comment:
Be careful with `string_view`. It copies the pointer. If `url` is destroyed too soon it points to nowhere!

Reply:
In all current use cases, the url is a `const char*` input to a method and all `string_view` are exclusively used inside that method. So we are safe.